executor/adapter.go

```diff
@@ -1283,22 +1282,15 @@ func (a *ExecStmt) GetTextToLog() string {
 	return sql
 }
 
-func (a *ExecStmt) observeStmtBeginForTopSQL() {
+func (a *ExecStmt) observeOnDigestKnownForTopSQL() {
 	if vars := a.Ctx.GetSessionVars(); variable.TopSQLEnabled() && vars.StmtStats != nil {
 		sqlDigest, planDigest := a.getSQLPlanDigest()
-		vars.StmtStats.OnExecutionBegin(sqlDigest, planDigest)
+		vars.StmtStats.OnDigestKnown(sqlDigest, planDigest)
 		// This is a special logic prepared for TiKV's SQLExecCount.
 		vars.StmtCtx.KvExecCounter = vars.StmtStats.CreateKvExecCounter(sqlDigest, planDigest)
```

Comment:
Is it possible to move this into `OnDigestKnown`?

Reply:
No, since `KvExecCounter ` is in `StmtCtx ` , not `StmtStats `

Reply:
I mean is it posssible to move it from `StmtCtx` into `vars.StmtStats`? As you can see, this PR did a hard work to maintain the semantic of "execution", and looks like it is reliable enough now for KvExecCounter to be placed inside.